contracts/voting-verifier/src/query.rs

```diff
@@ -1,28 +1,256 @@
+use axelar_wasm_std::voting::{Poll, PollStatus};
 use connection_router::state::{CrossChainId, Message};
+use cosmwasm_schema::cw_serde;
 use cosmwasm_std::{Deps, StdResult};
 
 use crate::error::ContractError;
-use crate::state::VERIFIED_MESSAGES;
+use crate::state::{self, POLLS, POLL_MESSAGES};
 
-pub fn verification_statuses(
-    deps: Deps,
-    messages: Vec<Message>,
-) -> StdResult<Vec<(CrossChainId, bool)>> {
+#[cw_serde]
+pub enum VerificationStatus {
+    Verified,
+    NotVerified,
+    InProgress, // still in an open poll
+    None,       // not in a poll
+}
+
+pub fn is_verified(deps: Deps, messages: &[Message]) -> StdResult<Vec<(CrossChainId, bool)>> {
     messages
-        .into_iter()
+        .iter()
         .map(|message| {
-            is_message_verified(deps, &message).map(|verified| (message.cc_id, verified))
+            verification_status(deps, message).map(|status| match status {
+                VerificationStatus::Verified => (message.cc_id.to_owned(), true),
+                _ => (message.cc_id.to_owned(), false),
+            })
         })
         .collect::<Result<Vec<(_, _)>, _>>()
         .map_err(Into::into)
 }
 
-pub fn is_message_verified(deps: Deps, message: &Message) -> Result<bool, ContractError> {
-    match VERIFIED_MESSAGES.may_load(deps.storage, &message.cc_id)? {
-        Some(stored) if stored != *message => {
+pub fn verification_status(
+    deps: Deps,
+    message: &Message,
+) -> Result<VerificationStatus, ContractError> {
+    match POLL_MESSAGES.may_load(deps.storage, &message.cc_id)? {
+        Some(stored) if stored.msg != *message => {
```

Comment:
I think this could be problematic. Invalid messages submitted to the verifier that have the same id but different contents than a valid message will block validation and routing for the valid message. We need to somehow allow for two different messages with the same id to be in polls at the same time.